### PR TITLE
QL: Start of the AST for `kind: predicates` evaluator logs

### PR DESCRIPTION
**Pull request by MathiasVP**

### CODE REVIEW

File: ql/ql/src/codeql_ql/StructuredLogs.qll [hunk 1]
```diff
@@ -213,14 +213,43 @@ module KindPredicatesLog {
...
 
+  class PipeLineRun extends Object {
+    PipeLineRuns runs;
+    int index;
+
+    PipeLineRun() { runs.getObject(index) = this }
+
+    PipeLineRuns getArray() { result = runs }
+
+    string getRaReference() { result = this.getString("raReference") }
```
Comment (tausbn) [inline ql/ql/src/codeql_ql/StructuredLogs.qll:232]: This one also goes against the style guide.

**Review by tausbn — DISMISSED** [review]
Two minor suggestions, otherwise LGTM. 👍

File: ql/ql/src/codeql_ql/StructuredLogs.qll (hunk 1 above)
Comment (MathiasVP) [reply ql/ql/src/codeql_ql/StructuredLogs.qll]: Thanks! Fixed in f155f19725bfe9930796f9da24b743afdf111f92.

File: ql/ql/src/codeql_ql/StructuredLogs.qll [hunk 2]
```diff
@@ -126,6 +126,101 @@ module EvaluatorLog {
...
+
+  abstract class SummaryEvent extends Object {
+    string evaluationStrategy;
+
+    bindingset[evaluationStrategy]
+    SummaryEvent() { evaluationStrategy = this.getString("evaluationStrategy") }
+
+    string getEvaluationStrategy() { result = evaluationStrategy }
+
+    string getRaHash() { result = this.getString("raHash") }
```
[the remark this answers is not in the document]
Comment (MathiasVP) [reply ql/ql/src/codeql_ql/StructuredLogs.qll]: Thanks! I've pushed a fix f155f19725bfe9930796f9da24b743afdf111f92 🙇

File: ql/ql/src/codeql_ql/StructuredLogs.qll [hunk 3]
```diff
@@ -126,6 +126,149 @@
...
+    int getResultSize() { result = this.getNumber("resultSize") }
+  }
+
+  class SentinelEmpty extends SummaryEvent {
+    SentinelEmpty() { evaluationStrategy = "SENTINEL_EMPTY" }
+  }
+
+  class PipeLineRun extends Object {
+    PipeLineRuns runs;
+    int index;
```
[the remark this answers is not in the document]
Comment (MathiasVP) [reply ql/ql/src/codeql_ql/StructuredLogs.qll]: Fixed in d64d03ec483eb779816bb998f97d98363262a368.

**Review by tausbn — APPROVED** [review]

**Pull request merged**